doc/lint-lifecycle.md

```diff
@@ -0,0 +1,66 @@
+# The Life of a Lint
+
+## Lint States
+
+### Proposed
+
+A lint begins its life as a **proposal** which is _pending_ until _accepted_.
+
+### Accepted
+
+After discussion and sufficient agreement within the team, a proposed lint becomes **accepted**.
```

Comment:
Do we want to say something here about details being hashed out? A problem with some of the older lint rules (and maybe some newer 🤣 ) is that they're really underspecified when they're implemented. So then people start using them, and raise new issues, which require whole new cycles of discussion, which might result in certain behavior being called false positives or false negatives, which are then again expensive to fix.

Alternatively, this could be a requirement for a lint moving to **stable**. WDYT?

Reply:
Yeah.  The truth is this is pretty ad hoc.  I'm not sure how formal we want to make it but I agree that getting more crisp about defining semantics *before* landing/accepting a lint would be good in general.  In practice, maybe this means prototyping and we might encourage that as legitimate too.

Reply:
I'd like to see a fully specified design be part of the criteria for moving to accepted.

But even if we do that, there will still be questions that arise after people start using the lint, so it probably needs to also be a requirement for moving to stable.